apps/opencs/model/tools/classcheck.cpp

```diff
@@ -37,26 +36,27 @@ void CSMTools::ClassCheckStage::perform (int stage, CSMDoc::Messages& messages)
 
     // A class should have a name
     if (class_.mName.empty())
-        messages.push_back (std::make_pair (id, class_.mId + " doesn't have a name"));
+    {
+        if (class_.mData.mIsPlayable != 0)
+            messages.add(id, "Name of a playable class is missing", "", CSMDoc::Message::Severity_Error);
+        else
+            messages.add(id, "Name is missing", "", CSMDoc::Message::Severity_Warning);
+    }
 
     // A playable class should have a description
     if (class_.mData.mIsPlayable != 0 && class_.mDescription.empty())
-        messages.push_back (std::make_pair (id, class_.mId + " doesn't have a description and it's playable"));
+        messages.add(id, "Description of a playable class is missing", "", CSMDoc::Message::Severity_Warning);
 
     // test for invalid attributes
     for (int i=0; i<2; ++i)
         if (class_.mData.mAttribute[i]==-1)
         {
-            std::ostringstream stream;
-
-            stream << "Attribute #" << i << " of " << class_.mId << " is not set";
-
-            messages.push_back (std::make_pair (id, stream.str()));
+            messages.add(id, "Attribute #" + std::to_string(i) + " is not set", "", CSMDoc::Message::Severity_Error);
         }
 
     if (class_.mData.mAttribute[0]==class_.mData.mAttribute[1] && class_.mData.mAttribute[0]!=-1)
     {
-        messages.push_back (std::make_pair (id, "Class lists same attribute twice"));
+        messages.add(id, "Same attribute is listed twice", "", CSMDoc::Message::Severity_Warning);
```

Comment:
I would consider this an error. Haven't test it, but it would expect having the same attribute listed twice to result in some weirdness when playing.